cheroot/server.py

```diff
@@ -1968,14 +1977,41 @@ def bind_unix_socket(self, bind_addr):  # noqa: C901  # FIXME
         return sock
 
     @staticmethod
-    def prepare_socket(bind_addr, family, type, proto, nodelay, ssl_adapter):
+    def _make_socket_reusable(socket_):
+        # Most BSD kernels implement SO_REUSEPORT the way that only the
+        # latest listener can read from socket. Some of BSD kernels also
+        # have SO_REUSEPORT_LB that works similarly to SO_REUSEPORT
+        # in Linux.
+        if hasattr(socket, 'SO_REUSEPORT_LB'):
+            socket_.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEPORT_LB, 1)
+        elif hasattr(socket, 'SO_REUSEPORT'):
+            socket_.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEPORT, 1)
+        elif IS_WINDOWS:
+            socket_.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        else:
+            raise NotImplementedError(
+                'Current platform does not support port reuse',
+            )
+
+    @classmethod
+    def prepare_socket(
+            cls, bind_addr, family, type, proto, nodelay, ssl_adapter,
+            reuse_port=False,
```

Comment:
I don't love how much plumbing was required for this change. It's also technically backward-incompatible in the case where a subclass has overridden `prepare_socket` but isn't accepting arbitrary `args` and `kwargs`.

That's not your fault, of course, but a problem with the current conventions/approach.

I'm tempted to say we should first (or after) refactor to make `prepare_socket` an instance method and remove the need to pass instance attributes through to it.

I'll plan to deal with that separately.